cmd/internal.go

```diff
@@ -335,17 +335,31 @@ func askMultiTarget(sess *session.Session, region string) (targets, domains []st
 func findInstances(sess *session.Session, region string) (map[string][]string, error) {
 	svc := ec2.New(sess, aws.NewConfig().WithRegion(region))
 
-	input := &ec2.DescribeInstancesInput{
-		Filters: []*ec2.Filter{{Name: aws.String("instance-state-name"), Values: []*string{aws.String("running")}}},
-	}
+	var ec2InstanceIds []string                    // used in the DescribeInstances call to filter results
+	var nonEc2Instances []*ssm.InstanceInformation // used to display any non-EC2 managed instances
 
-	// get managed instances
-	insts, err := findManagedInstances(sess, region)
-	if err != nil || len(insts) == 0 {
-	} else { // if instances exist.
-		input.Filters = append(input.Filters, &ec2.Filter{Name: aws.String("instance-id"), Values: aws.StringSlice(insts)})
+	managedInstances, err := findManagedInstances(sess, region) // get all ssm connected instances
+	if err != nil || len(managedInstances) == 0 {
```

Comment:
@tedsmitt 
A one of the cases raised errors is to lack  "ssm:DescribeInstanceInformation" on AWS permissions.

In a specific situation, "ssm:DescribeInstanceInformation" permmision is redundant.

So if err ! =nil,  currently bypasses a filter such as aws.String("instance-id").

Your commit doesn't show instances list to users without "ssm:DescribeInstanceInformation".

Reply:
Thanks @gjbae1212, I'll amend and test this now